lib/providers/accounts_provider.dart

```diff
@@ -11,11 +11,18 @@ final accountColorProvider = StateProvider<int>((ref) => 0);
 final accountMainSwitchProvider = StateProvider<bool>((ref) => false);
 final countNetWorthSwitchProvider = StateProvider<bool>((ref) => true);
 final selectedAccountCurrentMonthDailyBalanceProvider = StateProvider<List<FlSpot>>((ref) => const []);
+final filterAccountProvider = StateProvider<Map<int, bool>>((ref) => {});
+
 
 class AsyncAccountsNotifier extends AsyncNotifier<List<BankAccount>> {
   @override
   Future<List<BankAccount>> build() async {
     ref.watch(mainAccountProvider.notifier).state = await _getMainAccount();
+    _getAccounts().then((value) => {
```

Comment:
I think the _getAccounts() call is redundant, you can just put a return inside the first one after the for loop, it should work.

```
return _getAccounts().then((value) => {
      for(BankAccount account in value) {
        ref.watch(filterAccountProvider.notifier).state[account.id!] = true
      }
      return value;
});
```